spec/controllers/embed_controller_spec.rb

```diff
@@ -23,8 +23,20 @@
       controller.request.stubs(:referer).returns('http://eviltrout.com/some-page')
     end
 
+    it "is successful if limit the setting is zero" do
+      SiteSetting.embed_link_to_top = false
+      topic = Fabricate(:topic)
+      SiteSetting.embed_post_limit = 0
+
+      get :comments, topic_id: topic.id
+      expect(response).to be_success
+    end
+
```

Comment:
I think there should be a spec that makes sure it returns the right URL depending the value of `embed_link_to_top`.